src/main.py

```diff
@@ -85,30 +104,68 @@ def log_speed(
     # Return the current time and reset the counter to zero
     return time.time(), 0
 
-async def insert_data(batch: list[dict], error_queue:Queue):
-    try:
-        highscores:list[dict] = [msg.get("hiscores") for msg in batch]
-        players:list[dict] = [msg.get("player") for msg in batch]
 
-        highscores = [playerHiscoreDataSchema(**hs) for hs in highscores if hs]
-        highscores = [hs.model_dump(mode="json") for hs in highscores ]
+async def insert_data(batch: list[dict], error_queue: Queue):
+    session: AsyncSession = await get_session()
 
-        session: AsyncSession = await get_session()
-        
-        logger.info(f"Received: {len(players)=}, {len(highscores)=}")
+    try:
+        # Transform the old data format into the new format
+        batch = [await transform_data(msg, session) for msg in batch]
+
+        scraper_data_list: list[dict] = [msg.get("scraper_data") for msg in batch]
+        player_skills_list: list[dict] = [msg.get("player_skills") for msg in batch]
+        player_activities_list: list[dict] = [
+            msg.get("player_activities") for msg in batch
+        ]
+        players: list[dict] = [msg.get("player") for msg in batch]
+
+        scraper_data_list = [ScraperDataCreate(**sd) for sd in scraper_data_list if sd]
+        player_skills_list = [
+            PlayerSkillCreate(**ps) for ps in player_skills_list if ps
+        ]
+        player_activities_list = [
+            PlayerActivityCreate(**pa) for pa in player_activities_list if pa
+        ]
+        players = [PlayerCreate(**p) for p in players if p]
+
+        logger.info(
+            f"Received: players={len(players)}, scraper_data={len(scraper_data_list)}, skills={len(player_skills_list)}, activities={len(player_activities_list)}"
+        )
 
         # start a transaction
         async with session.begin():
-            # insert into table values ()
-            insert_sql:Insert = insert(PlayerHiscoreData)
-            insert_sql = insert_sql.values(highscores)
-            insert_sql = insert_sql.prefix_with("ignore")
-            await session.execute(insert_sql)
-            # update table

```

Comment:
we need to both suppor the old & new flow maybe adding an extra function would help